docs(aws-kms-tls-auth): add readme (#5409)
aws/s2n-tls: bindings/rust/aws-kms-tls-auth/DEVELOPMENT.md

# Managing Algorithm & Format Changes.

## Backwards Compatibility: Required

> Changes must always be backwards compatible. More specifically, server's must always be able to deserialize earlier versions of PskIdentities, otherwise all of the in-flight communications would fail when an upgrade happens

Reviewer: ```suggestion
Changes must always be backwards compatible. More specifically, server's must always be able to deserialize earlier versions of PskIdentities, otherwise all of the in-flight communications would fail when an upgrade happens.
```

> Changes must always be backwards compatible. More specifically, server's must always be able to deserialize earlier versions of PskIdentities, otherwise all of the in-flight communications would fail when an upgrade happens.
> 

Reviewer 2: ```suggestion
Changes must always be backwards compatible. More specifically, servers must always be able to deserialize earlier versions of PskIdentities, otherwise all of the in-flight communications would fail when an upgrade happens.

```

## Forward Compatibility: Customer Responsibility

> We generally do not promise forwards compatibility: A `0.0.1`, V1 enabled server might not be able to handshake with a `0.0.2` V2 enabled client. We will strive to maintain forward compatibility, but if there was ever an upgrade from `AES_256_GCM_SIV` to `AES_512_GCM_SIV`, that would not be forward compatible. 
> 

Reviewer 2: ```suggestion
We generally do not promise forwards compatibility: A `0.0.1` V1 enabled server might not be able to handshake with a `0.0.2` V2 enabled client. We will strive to maintain forward compatibility, but if there was ever an upgrade from `AES_256_GCM_SIV` to `AES_512_GCM_SIV`, that would not be forward compatible. 

```

> It would be the customer responsibility for first deploy version 0.0.2 to all servers, and only then would it be safe to enabled `PskVersion::V2`. For further

Reviewer: ```suggestion
It would be the customer's responsibility to first deploy version 0.0.2 to all servers, and only then would it be safe to enable `PskVersion::V2`. For further
```
